File: Frontend/src/app/domain/RegistrationToken.ts
export class RegistrationToken {
    id: number;
    token: string;
    validDuration: number;
    email: string;
    createdBy: number;

    constructor(id: number, token: string, validDuration: number, email: string, createdBy: number) {
        this.id = id;
        this.token = token;
        this.validDuration = validDuration;
        this.email = email;
        this.createdBy = createdBy;
    }
}

File: Auth/src/main/java/com/example/w7pg2_auth/entity/RegistrationToken.java
package com.example.w7pg2_auth.entity;

import java.io.Serializable;
import java.sql.Timestamp;
import java.time.LocalDateTime;
import java.util.Calendar;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.*;

@Entity
@Table(name="RegistrationToken")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class RegistrationToken implements Serializable {
	public static final int EXPIRATION = 60 * 3;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	int id;
	
	@Column(name = "token")
	String token;
	
	@Column(name = "validDuration")
	String validDuration;
	
	@Column(name = "email")
	String email;
	
	@OneToOne
	@JoinColumn(name="createdBy")
	Employee createdBy;
}
